src/test/test_ar_process.py

```diff
@@ -12,33 +13,73 @@ def test_ar_can_be_sampled():
     Check that an AR process
     can be initialized and sampled from
     """
-    ar1 = ARProcess("arprocess", 5, jnp.array([0.95]), jnp.array([0.5]))
+    ar1 = ARProcess("ar1process")
     with numpyro.handlers.seed(rng_seed=62):
-        # can sample with and without inits
-        ar1(duration=3532, inits=jnp.array([50.0]))
-        ar1(duration=5023)
+        # can sample
+        ar1(
+            n=3532,
+            init_vals=jnp.array([50.0]),
+            autoreg=jnp.array([0.95]),
+            noise_sd=0.5,
+        )
+
+    ar3 = ARProcess("ar3process")
 
-    ar3 = ARProcess(
-        "arprocess", 5, jnp.array([0.05, 0.025, 0.025]), jnp.array([0.5])
-    )
     with numpyro.handlers.seed(rng_seed=62):
-        # can sample with and without inits
-        ar3(duration=1230)
-        ar3(duration=52, inits=jnp.array([50.0, 49.9, 48.2]))
+        # can sample
+        ar3(
+            n=1230,
+            init_vals=jnp.array([50.0, 49.9, 48.2]),
+            autoreg=jnp.array([0.05, 0.025, 0.025]),
+            noise_sd=0.5,
+        )
+        ar3(
+            n=1230,
+            init_vals=jnp.array([50.0, 49.9, 48.2]),
+            autoreg=jnp.array([0.05, 0.025, 0.025]),
+            noise_sd=[0.25],
+        )
+        ar3(
+            n=1230,
+            init_vals=jnp.array([50.0, 49.9, 48.2]),
+            autoreg=jnp.array([0.05, 0.025, 0.025]),
+            noise_sd=jnp.array([0.25]),
+        )
```

Comment:
opportunities to DRY here

Reply:
Agree but I think it can be a separate PR.

Reply:
Created #405